src/main/java/com/project/stockexchangeappbackend/service/StockServiceImpl.java

```diff
@@ -10,34 +11,59 @@
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import javax.persistence.EntityExistsException;
 import javax.persistence.EntityNotFoundException;
 
 @Service
 @RequiredArgsConstructor
 public class StockServiceImpl implements StockService {
 
-    private final StockRepository repository;
+    private final StockRepository stockRepository;
 
     @Override
     @LogicBusinessMeasureTime
     @Transactional(readOnly = true)
     public Stock getStockById(Long id) {
-        return repository.findById(id).orElseThrow(() -> new EntityNotFoundException("Stock Not Found"));
+        return stockRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Stock Not Found"));
     }
 
     @Override
     @LogicBusinessMeasureTime
     @Transactional(readOnly = true)
     public Page<Stock> getStocks(Pageable pageable, Specification<Stock> specification) {
-        return repository.findAll(specification, pageable);
+        return stockRepository.findAll(specification, pageable);
     }
 
     @Override
     @LogicBusinessMeasureTime
+    @Transactional
     public Stock getStockByAbbreviation(String abbreviation) {
-        return repository.findByAbbreviationIgnoreCase(abbreviation).orElseThrow(() ->
+        return stockRepository.findByAbbreviationIgnoreCase(abbreviation).orElseThrow(() ->
                 new EntityNotFoundException("Stock Not Found"));
     }
 
+    @Override
+    @LogicBusinessMeasureTime
+    public void updateStock(StockDTO stockDTO, String id) {
+        if (stockRepository.findByNameIgnoreCase(stockDTO.getName()).isPresent()) {
+            throw new EntityExistsException("Stock with given name already exists");
+        }
+        if (stockRepository.findByAbbreviationIgnoreCase(stockDTO.getAbbreviation()).isPresent()) {
+            throw new EntityExistsException("Stock with given abbreviation already exists");
+        }
```

Comment:
it will be better i you use findByNameIgnoreCaseOrAbbreviationIgnoreCase. Then you have just one request to db

Reply:
and remember about trimming here as well